compiler/codegen/CodeGenRA.cpp

```diff
@@ -407,6 +407,15 @@ OMR::CodeGenerator::prepareRegistersForAssignment()
       if (registerCursor->getKind() != TR_SSR)
          registerCursor->setFutureUseCount(registerCursor->getTotalUseCount());
       km = registerCursor->getKindAsMask();
+
+      /*
+       * Sometimes real registers marked as being a TR_FPR register can be used as a TR_VRF register
+       * isFPRUsedAsVFR is used to check for this
+       * if true, RA is told is may need to assign TR_VRF as well as TR_FPR registers if a TR_FPR real register is seen
+       */
+      if (registerCursor->getKind() == TR_FPR && self()->isFPRUsedAsVFR())
+         km |= TO_KIND_MASK(TR_VRF);
+
```

Comment:
Can you elaborate on why the `isFPRUsedAsVFR()` query is needed and what the effect its trying to achieve?  What this seems to imply is that if you encounter an FPR on x86 you will (for all intents and purposes) allow TR_VFRs to assign as well.  This seems to be the case even if you aren't doing autoSIMD or if there are any TR_VRFs in use in this method.